flexget/plugins/api_t411.py

```diff
@@ -0,0 +1,648 @@
+                        # Parsing & indexing terms
+                        if term_id not in terms:
+                            term = Term(id=int(term_id), name=term_name)
+                            term_type.terms.append(term)
+
+        return category_to_term_type, term_types
+
+    @staticmethod
+    def map_search_result_entry(json_entry, download_auth=None):
+        """
+        Parse json object of a torrent entry to flexget Entry
+        :param download_auth: Requests authenticator
+        """
+        result = Entry()
+        result['t411_torrent_id'] = json_entry['id']
+        result['title'] = json_entry['name']
+        result['url'] = T411RestClient.download_url(json_entry['id'])
+        result['t411_category'] = int(json_entry['category'])
+        result['seeders'] = int(json_entry['seeders'])
+        result['leechers'] = int(json_entry['leechers'])
+        result['t411_comments'] = int(json_entry['comments'])
+        result['t411_verified'] = json_entry['isVerified'] is '1'
+        result['t411_pubdate'] = datetime.strptime(json_entry['added'], T411ObjectMapper.date_format)
+        result['content_size'] = int(json_entry['size']) / (1024 ** 2)
+        result['t411_times_completed'] = int(json_entry['times_completed'])
+        result['t411_category_name'] = json_entry['categoryname']
+        result['t411_category_image'] = json_entry['categoryimage']
+        result['t411_privacy'] = json_entry['privacy']
+        result['t411_owner_id'] = int(json_entry['owner'])
+        result['t411_owner_username'] = json_entry['username']
+        result['download_auth'] = download_auth
+        return result
+
+    @staticmethod
+    def map_details(json_details, resolver):
+        result = Torrent()
+        result.id = json_details.get('id')
+        result.name = json_details.get('name')
+        result.category_id = json_details.get('category')
+
+        for (term_type_name, terms_candidat) in json_details.get('terms').iteritems():
+            if isinstance(terms_candidat, list):
+                for term_name in terms_candidat:
+                    result.terms.append(resolver(result.category_id, term_type_name, term_name))
+            else:
+                result.terms.append(resolver(result.category_id, term_type_name, terms_candidat))
+
+        return result
+
+
+def cache_required(func):
+    """
+    Decorator for ensuring cached data into db.
+    If not a synchronize will be launched
+    :param func:
+    :return:
+    """
+
+    def wrapper(self, *args, **kwargs):
+        log.debug("Checking database...")
+        if not self.has_cached_criterias():
+            log.debug('None cached data. Synchronizing...')
+            self.synchronize_database()
+        return func(self, *args, **kwargs)
+
+    return wrapper
+
+
+class T411Proxy(object):
+    """
+    A T411 proxy service. This proxy interact both with
+    T411 Rest Client and T411 local database.
+    """
+
+    @with_session
```

Comment:
I think you should wrap each individual method that needs a db session in with_session, rather than using self.session